fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryTest.java

```diff
@@ -233,4 +241,62 @@ private void assertMetricCountAtMost(StoreTimer.Event metric, int maximumValue)
         assertThat(String.format("Metric %s should be called at most %d times", metric, maximumValue),
                 timer.getCount(metric), lessThanOrEqualTo(maximumValue));
     }
+
+
+    @Test
+    void testFileLock() throws IOException {
+        String lockName = "file.lock";
+        final Lock lock1 = directory.obtainLock(lockName);
+        lock1.ensureValid();
+        RecordCoreException e = assertThrows(RecordCoreException.class, () -> directory.obtainLock(lockName));
+        assertTrue(e.getMessage().contains("found old lock"));
+        lock1.ensureValid();
+        lock1.close();
+        assertThrows(AlreadyClosedException.class, lock1::ensureValid);
+        final Lock lock2 = directory.obtainLock(lockName);
+        lock2.ensureValid();
+        e = assertThrows(RecordCoreException.class, () -> directory.obtainLock(lockName));
+        assertTrue(e.getMessage().contains("found old lock"));
+        lock2.ensureValid();
+        lock2.close();
+    }
+
+    @Test
+    void testAgilityContext() throws ExecutionException, InterruptedException {
```

Comment:
It would be good to have a test of the atomicity of `apply`/`accept`. You may have better ideas, but the first one that jumps into my mind is to have a bunch of threads that do something like:
```
context.apply(aContext -> {
    value = randomBytes()
    aContext.set(key1,  value);
    aContext.set(key2, value);
});
```
and a bunch of other threads doing:
```
context.accept(aContext -> {
    assertEquals(aContext.get(key2), aContext.get(key1)); // intentionally in the opposite order
});
```

Reply:
Probably also worth testing that the non-agile one's atomicity, namely that a series of `get`/`set` interspersed should not be seeing other contexts work.

Reply:
✅